requirements.txt

```diff
@@ -1,12 +1,16 @@
 # https://github.com/mediawiki-utilities/python-mwxml/pull/19
 git+https://github.com/gdedrouas/python-mwxml@xml_format_0.11
 git+https://github.com/mediawiki-utilities/python-mwtypes@updates_schema_0.11
-chainlit==1.1.202
+chainlit==1.1.401
 chromadb>=0.4.22
 discord>=2.3.2
 fastapi>=0.109.1
-langchain==0.2.5
-langchain-community==0.2.5
+langchain==0.2.12
+langchainhub>=0.1.20
+langchain-chroma>=0.1.2
+langchain-community==0.2.11
+langchain-huggingface>=0.0.3
+langchain-ollama==0.1.1

```

Comment:
Langchain is very overkill with the pypi packages, but this is how you avoid kitchen-sink.